DuckDuckGo/OnboardingExperiment/ContextualOnboarding/ContextualOnboardingPresenter.swift

```diff
@@ -0,0 +1,131 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+import Core
+
+protocol ContextualOnboardingPresenting {
+    func presentContextualOnboarding(for spec: DaxDialogs.BrowsingSpec, in vc: TabViewControllerType)
+}
+
+final class ContextualOnboardingPresenter: ContextualOnboardingPresenting {
+    private let variantManager: VariantManager
+    private let daxDialogsFactory: ContextualDaxDialogsFactory
+
+    init(variantManager: VariantManager, daxDialogsFactory: ContextualDaxDialogsFactory = ExistingLogicContextualDaxDialogsFactory()) {
+        self.variantManager = variantManager
+        self.daxDialogsFactory = daxDialogsFactory
+    }
+
+    func presentContextualOnboarding(for spec: DaxDialogs.BrowsingSpec, in vc: TabViewControllerType) {
+        if variantManager.isSupported(feature: .newOnboardingIntro) {
+            presentExperimentContextualOnboarding(for: spec, in: vc)
+        } else {
+            presentControlContextualOnboarding(for: spec, in: vc)
+        }
+    }
+
+}
+
+// MARK: - Private
+
+private extension ContextualOnboardingPresenter {
+
+    func presentControlContextualOnboarding(for spec: DaxDialogs.BrowsingSpec, in vc: TabViewControllerType) {
+        vc.performSegue(withIdentifier: "DaxDialog", sender: spec)
+    }
+
+    func presentExperimentContextualOnboarding(for spec: DaxDialogs.BrowsingSpec, in vc: TabViewControllerType) {
+
+        func animate(daxController: UIViewController, visible isVisible: Bool, onCompletion: ((Bool) -> Void)? = nil) {
+            daxController.view.isHidden = !isVisible
+            UIView.animate(
+                withDuration: 0.3,
+                animations: {
+                    daxController.view.alpha = isVisible ? 1 : 0
+                    daxController.parent?.view.layoutIfNeeded()
+                },
+                completion: onCompletion
+            )
+        }
+
+        // Before presenting a new dialog, remove any existing ones.
+        vc.daxDialogsStackView.arrangedSubviews.filter({ $0 != vc.webViewContainerView }).forEach {
+            vc.daxDialogsStackView.removeArrangedSubview($0)
+            $0.removeFromSuperview()
+        }
+
+        // Ask the Dax Dialogs Factory for a view for the given spec
+        let controller = daxDialogsFactory.makeView(for: spec) { [weak vc] in
+            guard let vc, let daxController = vc.daxContextualOnboardingController else { return }
+
+            // Collapse stack view and remove dax controller
+            animate(daxController: daxController, visible: false) { _ in
+                vc.daxDialogsStackView.removeArrangedSubview(daxController.view)
+                vc.removeChild(daxController)
+            }
+        }
+        controller.view.isHidden = true
+        controller.view.alpha = 0
```

Comment:
AFAIR I hadn't set the alpha to animate the appearance/disappearance of a view in a stack view. However, without it, the view simply appears without any animation. The disappearance works fine without the alpha 😕